### PR TITLE
Populate curatingInstitutions in ResourceService->createPublication #NP-46342

### PR DESCRIPTION
**Pull request by torbjokv**
- Reduce number of places services is constructed with getMessageService, getTicketService and getResourceServiceBuilder
- Add support for empty entityDescription when using CuratingInstitutionsUtil
- Add deprication notice on insertPreexistingPublication to discourage use
- Add setCuratingInstitutions in insertResource to populate data

### CODE REVIEW

File: cristin-import/src/test/java/no/unit/nva/cristin/lambda/CristinEntryEventConsumerTest.java [hunk 1]
```diff
@@ -114,7 +114,7 @@ class CristinEntryEventConsumerTest extends AbstractCristinImportTest {
     @BeforeEach
     public void init() {
         super.init();
-        resourceService = ResourceService.builder().withDynamoDbClient(super.client).build();
+        resourceService =getResourceServiceBuilder().build();
```
Comment (brinxmat) [inline cristin-import/src/test/java/no/unit/nva/cristin/lambda/CristinEntryEventConsumerTest.java:117]: Formatting — maybe time to use 

```
plugins {
	id 'com.diffplug.spotless' version "6.25.0"
}
```

File: expansion/src/test/java/no/unit/nva/expansion/ResourceExpansionServiceTest.java [hunk 2]
```diff
@@ -615,7 +615,7 @@ private ResourceExpansionServiceImpl expansionServiceReturningNviCandidate(Publi
         when(response.body()).thenReturn(responseBody);
         when(mock.fetchResponse(fetchNviCandidateUri(publication), "application/json")).thenReturn(Optional.of(response));
         return new ResourceExpansionServiceImpl(getResourceServiceBuilder().build(),
-                                                            new TicketService(client),
+                                                            new TicketService(client, mock),
```
Comment (brinxmat) [inline expansion/src/test/java/no/unit/nva/expansion/ResourceExpansionServiceTest.java:618]: We should do something about the naming of `mock`

File: messages/src/test/java/no/unit/nva/publication/messages/create/NewCreateMessageHandlerTest.java [hunk 3]
```diff
@@ -63,15 +61,13 @@ class NewCreateMessageHandlerTest extends ResourcesLocalTest {
...
 
     @BeforeEach
     public void setup() {
         super.init();
         this.resourceService = getResourceServiceBuilder().build();
-        this.ticketService = new TicketService(client);
-        this.uriRetriever = mock(UriRetriever.class);
-        MessageService messageService = new MessageService(client);
+        this.ticketService = getTicketService();
+        MessageService messageService = getMessageService();
```
Comment (brinxmat) [inline messages/src/test/java/no/unit/nva/publication/messages/create/NewCreateMessageHandlerTest.java:70]: `var`

File: publication-commons/src/main/java/no/unit/nva/publication/model/utils/CuratingInstitutionsUtil.java [hunk 4]
```diff
@@ -42,7 +47,9 @@ private static Stream<URI> getOrganizationIds(Contributor contributor) {
     }
 
     private static Stream<Contributor> getVerifiedContributors(EntityDescription entityDescription) {
-        return entityDescription.getContributors()
+        return Optional.ofNullable(entityDescription)
+                   .map(EntityDescription::getContributors)
+                   .orElse(Collections.emptyList())
```
Comment (brinxmat) [inline publication-commons/src/main/java/no/unit/nva/publication/model/utils/CuratingInstitutionsUtil.java:52]: We don't do this automatically in data-model?

File: publication-commons/src/main/java/no/unit/nva/publication/service/impl/ResourceService.java [hunk 5]
```diff
@@ -93,13 +95,15 @@ public class ResourceService extends ServiceWithTransactions {
     private final ReadResourceService readResourceService;
     private final UpdateResourceService updateResourceService;
     private final DeleteResourceService deleteResourceService;
+    private final UriRetriever uriRetriever;
 
     protected ResourceService(AmazonDynamoDB dynamoDBClient, String tableName, Clock clock,
-                              Supplier<SortableIdentifier> identifierSupplier) {
+                              Supplier<SortableIdentifier> identifierSupplier, UriRetriever uriRetriever) {
```
Comment (brinxmat) [inline publication-commons/src/main/java/no/unit/nva/publication/service/impl/ResourceService.java:101]: Can you format each argument on a new line?

File: publication-commons/src/main/java/no/unit/nva/publication/utils/RdfUtils.java [hunk 6]
```diff
@@ -41,7 +41,7 @@ public static URI getTopLevelOrgUri(UriRetriever uriRetriever, URI id) {
         var data = attempt(() -> uriRetriever.getRawContent(id, APPLICATION_JSON)).orElseThrow();
 
         if (data.isEmpty()) {
-            return id;
+            return null;
```
Comment (brinxmat) [inline publication-commons/src/main/java/no/unit/nva/publication/utils/RdfUtils.java:44]: Now wondering what the thinking for the previous iteration was…

Comment (torbjokv) [reply publication-commons/src/main/java/no/unit/nva/publication/utils/RdfUtils.java:44]: yes, me too. Maybe to make the unit tests easier? You could maybe argue that it's not completely wrong. My current level is the "most top" unit that I know of. :-P

Comment (torbjokv) [reply publication-commons/src/main/java/no/unit/nva/publication/utils/RdfUtils.java:44]: could be that an Exception would be better than null here also

File: publication-commons/src/main/java/no/unit/nva/publication/model/utils/CuratingInstitutionsUtil.java (hunk 4 above)
Comment (torbjokv) [reply publication-commons/src/main/java/no/unit/nva/publication/model/utils/CuratingInstitutionsUtil.java:52]: not sure. tried to search a bit. we do this several places for sure

Comment (brinxmat) [reply publication-commons/src/main/java/no/unit/nva/publication/model/utils/CuratingInstitutionsUtil.java:52]: Should probably look into the model issues at a deeper level

**Review by brinxmat — APPROVED** [review]

File: publication-commons/src/main/java/no/unit/nva/publication/model/utils/CuratingInstitutionsUtil.java (hunk 4 above)
Comment (torbjokv) [reply publication-commons/src/main/java/no/unit/nva/publication/model/utils/CuratingInstitutionsUtil.java:52]: will take this in this pr and then I can do fixes in next PR if needed

**Pull request merged**